test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -102,9 +109,246 @@ spec = do
+                         ]
+                       )
+                     , ( "Wildcards name"
+                       , wildcardsWalletName
+                       , [ expectResponseCode @IO HTTP.status202
+                         , expectFieldEqual walletName wildcardsWalletName
+                         ]
+                       )
+                     ]
+        forM_ matrix $ \(title, walName, expectations) -> it title $ \ctx -> do
+            let payload = Json [json| {
+                    "name": #{walName},
+                    "mnemonic_sentence": #{mnemonics24},
+                    "passphrase": "Secure Passphrase"
+                    } |]
+            r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+            verify r expectations
+
+    it "WALLETS_CREATE_04 - [] as name -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": [],
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected Text, encountered Array"
+            ]
+
+    it "WALLETS_CREATE_04 - Num as name -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": 123,
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected Text, encountered Number"
+            ]
+
+    it "WALLETS_CREATE_04 - Name param missing -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "key \"name\" not present"
+            ]
+
+    describe "WALLETS_CREATE_05 - Mnemonics" $ do
+            let matrix = [ ( "[] as mnemonic_sentence -> fail"
+                           , []
+                           , [ expectResponseCode @IO HTTP.status400
+                             , expectErrorMessage "ErrMnemonicWords (ErrWrongNumberOfWords 0 24)"
+                             ]
+                           )
+                         , ( "specMnemonicSentence -> fail"
+                           , specMnemonicSentence
+                           , [ expectResponseCode @IO HTTP.status400
+                             , expectErrorMessage "ErrMnemonicWords (ErrWrongNumberOfWords 15 24)"
+                             ] -- probably need to modify after bug #192 fixed
```

Comment:
Indeed. Let's get #192 resolved today and merge this one after in its corrected form :+1: